src/main/java/org/opencadc/scienceportal/SciencePortalAuthAction.java

```diff
@@ -142,6 +144,13 @@ protected Subject getCurrentSubject(final URL targetURL) throws Exception {
                            .forEach(authMethod -> subject.getPublicCredentials().remove(authMethod));
                     subject.getPublicCredentials().add(AuthMethod.TOKEN);
                 }
+            } else if (AuthenticationUtil.getAuthMethod(subject) == AuthMethod.COOKIE) {
+                final Set<SSOCookieCredential> publicCookieCredentials = subject.getPublicCredentials(SSOCookieCredential.class);
+                if (!publicCookieCredentials.isEmpty()) {
+                    final SSOCookieCredential publicCookieCredential = publicCookieCredentials.toArray(new SSOCookieCredential[0])[0];
+                    subject.getPublicCredentials().add(new SSOCookieCredential(publicCookieCredential.getSsoCookieValue(), targetURL.getHost(),
```

Comment:
Do entries in the `getPublicCredentials()` collection ever need cleaning up, as is done for tokens?

Reply:
They do not.  They exist only in the scope of the call to authenticate the user, then are removed.